plugin/aws/s3Bucket.go

```diff
@@ -198,7 +207,32 @@ func (b *s3Bucket) Metadata(ctx context.Context) (plugin.MetadataMap, error) {
 		return nil, err
 	}
 
-	metadata["region"] = b.region
+	region, err := b.getRegion(ctx)
+	if err != nil {
+		return nil, err
+	}
+	metadata["region"] = region
 
 	return metadata, nil
 }
+
+func (b *s3Bucket) getRegion(ctx context.Context) (string, error) {
+	var err error
+	b.regioner.Do(func() {
```

Comment:
I'd have this try again if getting the region returns an error (the retry would happen when it's invoked). You can do that by resetting the regioner in the `Do` function (I do something similar in https://github.com/puppetlabs/wash/blob/master/plugin/aws/ec2Instance.go#L217), and storing the region error in a `regionErr` variable.

Reply:
Updated.